### ISSUE TITLE
Implement `execution/evm.go`

### ISSUE THREAD

**Issue by star-gazer111** [BlocSoc-iitr/selene#29]

**Comment by x-senpai-x** [comment]
Plz assign this

**Comment by star-gazer111** [comment]
@x-senpai-x what are the updates on the progress?

**Comment by x-senpai-x** [comment]
I need execution client (execution.go) in evm.go and callOpts(in types.go) , so should I just assume that they have been defined , although it would be better if I can take a look at those fies first .

**Comment by star-gazer111** [comment]
> @star-gazer111 I need execution client (execution.go) and callOpts(in types.go) for evm.go, so should I just assume that they have been defined , although it would be better if I can take a look at those files first .

yes, for now. They'll be marked exportable in their respective PRs, I won't merge until then so u can reference later & change names/etc if some change is needed once they have opened their PRs.

**Comment by x-senpai-x** [comment]
@star-gazer111 I am only left with call functions and getEnv , but I am unable to find the equivalent go packages (For [ResultandState](https://github.com/a16z/helios/blob/master/execution/src/evm.rs#L57))

**Comment by star-gazer111** [comment]
> @star-gazer111 I am only left with call functions and getEnv , but I am unable to find the equivalent go packages (For [ResultandState](https://github.com/a16z/helios/blob/master/execution/src/evm.rs#L57))

can you check [here](https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go) please?
